src/object/particlesystem.cpp

```diff
@@ -99,8 +99,11 @@ ParticleSystem::draw(DrawingContext& context)
     pos.x = fmodf(particle->pos.x - scrollx, virtual_width);
     if ((pos.x + static_cast<float>(particle_width)) < 0) pos.x += virtual_width;
 
-    pos.y = fmodf(particle->pos.y - scrolly, virtual_height);
-    if (pos.y < 0) pos.y += virtual_height;
+
+    const float particle_height = static_cast<float>(particle->texture->get_height());
+    const float virtual_height_particle = this->virtual_height + particle_height;
```

Comment:
No need to use `this->`.

Reply:
No problem, I removed it.
Sorry, I am too accustomed to a bit different programming style and I wrote it "automatically" (I usually do it in my projects).